inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php

```diff
@@ -147,6 +147,14 @@ public function remove_used_css_from_html( string $html, array $unprocessed_css
 		$unprocessed_links  = $this->unprocessed_flat_array( 'link', $unprocessed_css );
 		$unprocessed_styles = $this->unprocessed_flat_array( 'inline', $unprocessed_css );
 
+		if ( ! $link_styles ) {
+			$link_styles = [];
+		}
+
+		if ( ! $inline_styles ) {
+			$inline_styles = [];
+		}
+
```

Comment:
@wp-media/php I'm thinking this kind of things indicates a flaw in the design of the method. the `find()` method could return an empty array instead of `false`. It has 2 advantages:
- A single return type
- No need for this type of conditional checks, since the variables will always be set to an iteratable value

What do you think?

Reply:
You are totally right, I will refactor this part to also use UrlTrait and make it return empty array if regex search failed.